### PR TITLE
Full Rails Engine

### PR DESCRIPTION
**Pull request by peterhicks**
This PR converts the `shiftcommerce-rails` plugin from a "frankenengine" gem to a full Rails engine, based upon the Rails engine scaffold.

This change should make testing and developing this gem much easier in future, as it now follows the expected and well-tested standard.

### CODE REVIEW

File: app/jobs/shift_commerce/application_job.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+module Shiftcommerce
+  class ApplicationJob < ActiveJob::Base
+  end
+end
```
Comment (ryantownsend) [inline app/jobs/shift_commerce/application_job.rb]: Remove this please – front-end websites shouldn't need background queues.

File: app/models/shift_commerce/application_record.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+module Shiftcommerce
+  class ApplicationRecord < ActiveRecord::Base
+    self.abstract_class = true
+  end
+end
```
Comment (ryantownsend) [inline app/models/shift_commerce/application_record.rb]: Remove this please – front-end websites shouldn't need ActiveRecord as models are loaded via the API.

File: app/jobs/shift_commerce/application_job.rb (hunk 1 above)
Comment (peterhicks) [inline app/jobs/shift_commerce/application_job.rb]: Oops. I removed the mailer, but looks like removing the job got lost while I was mucking around. – will do that now.

File: spec/dummy/app/assets/javascripts/cable.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+//= require action_cable
+//= require_self
+//= require_tree ./channels
+
+(function() {
+  this.App || (this.App = {});
+
+  App.cable = ActionCable.createConsumer();
+
+}).call(this);
```
Comment (ryantownsend) [inline spec/dummy/app/assets/javascripts/cable.js]: Remove this please – front-end websites shouldn't need or be using ActionCable

File: spec/dummy/app/channels/application_cable/channel.rb [hunk 4]
```diff
@@ -0,0 +1,4 @@
+module ApplicationCable
+  class Channel < ActionCable::Channel::Base
+  end
+end
```
Comment (ryantownsend) [inline spec/dummy/app/channels/application_cable/channel.rb]: Remove this please – front-end websites shouldn't need or be using ActionCable

File: spec/dummy/app/channels/application_cable/connection.rb [hunk 5]
```diff
@@ -0,0 +1,4 @@
+module ApplicationCable
+  class Connection < ActionCable::Connection::Base
+  end
+end
```
Comment (ryantownsend) [inline spec/dummy/app/channels/application_cable/connection.rb]: Remove this please – front-end websites shouldn't need or be using ActionCable

File: spec/dummy/app/jobs/application_job.rb [hunk 6]
```diff
@@ -0,0 +1,2 @@
+class ApplicationJob < ActiveJob::Base
+end
```
Comment (ryantownsend) [inline spec/dummy/app/jobs/application_job.rb]: Remove this please – front-end websites shouldn't need or be using ActiveJob

File: spec/dummy/bin/setup [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  system! 'gem install bundler --conservative'
+  system('bundle check') || system!('bundle install')
+
+  # puts "\n== Copying sample files =="
+  # unless File.exist?('config/database.yml')
+  #   cp 'config/database.yml.sample', 'config/database.yml'
+  # end
+
+  puts "\n== Preparing database =="
+  system! 'bin/rails db:setup'
```
Comment (ryantownsend) [inline spec/dummy/bin/setup]: Remove this please – front-end websites shouldn't using the database

File: spec/dummy/bin/update [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+chdir APP_ROOT do
+  # This script is a way to update your development environment automatically.
+  # Add necessary update steps to this file.
+
+  puts '== Installing dependencies =='
+  system! 'gem install bundler --conservative'
+  system('bundle check') || system!('bundle install')
+
+  puts "\n== Updating database =="
+  system! 'bin/rails db:migrate'
```
Comment (ryantownsend) [inline spec/dummy/bin/update]: Remove this please – front-end websites shouldn't be using the DB

File: spec/dummy/config/cable.yml [hunk 9]
```diff
@@ -0,0 +1,9 @@
+development:
+  adapter: async
+
+test:
+  adapter: async
+
+production:
+  adapter: redis
+  url: redis://localhost:6379/1
```
Comment (ryantownsend) [inline spec/dummy/config/cable.yml]: Remove this please – front-end websites shouldn't need or be using ActionCable

File: spec/dummy/config/application.rb [hunk 10]
```diff
@@ -0,0 +1,23 @@
+require_relative 'boot'
+
+# Pick the frameworks you want:
+# require "active_record/railtie"
+require "action_controller/railtie"
+require "action_view/railtie"
+# require "action_mailer/railtie"
+require "active_job/railtie"
```
Comment (ryantownsend) [inline spec/dummy/config/application.rb]: Remove this line please – front-end websites shouldn't need or be using ActiveJob

File: spec/dummy/config/application.rb [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+# require "action_cable/engine"
+# require "rails/test_unit/railtie"
+require "sprockets/railtie"
+
+Bundler.require(*Rails.groups)
+require "shiftcommerce"
```
Comment (ryantownsend) [inline spec/dummy/config/application.rb]: Shoudn't this be `require 'shiftcommerce-rails'`?

File: spec/dummy/app/views/layouts/application.html.erb [hunk 12]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>Dummy</title>
+    <%= csrf_meta_tags %>
+
+    <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload' %>
+    <%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>
```
Comment (ryantownsend) [inline spec/dummy/app/views/layouts/application.html.erb]: Drop the turbolinks stuff, this is only the dummy but we shouldn't be using it.

File: app/controllers/shift_commerce/application_controller.rb [hunk 13]
```diff
@@ -0,0 +1,5 @@
+module TestEngine
```
Comment (ryantownsend) [inline app/controllers/shift_commerce/application_controller.rb]: Wrong module name

File: bin/rails [hunk 14]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env ruby
+# This command will automatically be run when you run "rails" with Rails gems
+# installed from the root of your application.
+
+ENGINE_ROOT = File.expand_path('../..', __FILE__)
+ENGINE_PATH = File.expand_path('../../lib/shiftcommerce/engine', __FILE__)
```
Comment (ryantownsend) [inline bin/rails]: Should this path have an underscore in `shiftcommerce`?

File: spec/dummy/config/routes.rb [hunk 15]
```diff
@@ -0,0 +1,3 @@
+Rails.application.routes.draw do
+  mount Shiftcommerce::Engine => "/shiftcommerce"
```
Comment (ryantownsend) [inline spec/dummy/config/routes.rb]: `Shiftcommerce::Engine` namespace needs capital 'C'

**Pull request merged**